### ISSUE TITLE
Display current line background color on normal mode

### ISSUE THREAD

**Issue by ctwdtw** [mickael-menu/ShadowVim#42]
Support Cursor and Current Line Color would be great

**Comment by ctwdtw** [comment]
When Using Xcode code built-in vim in normal mode, the current line will be highlighted with the color set through `Xcode` -> `Settings` -> `Thems` -> `Source Edtor` -> `Current Line`. 

However, it seems that ShadowVim does not support such highlighted colors. 
It would be great to have `cursor color` and `current line color` supported in ShadowVim.

Xcode Built-in Vim:
![XCode-build-in-vim-has-current-line-bakcground-highlight](https://user-images.githubusercontent.com/7893446/227174469-a7abc17d-dc47-4cdf-b0cc-2935f0a3990f.png)


Shadow Vim:
![Shadow-vim-has-no-current-line-bakcground-highlight](https://user-images.githubusercontent.com/7893446/227173307-1eddf43a-8ff2-496b-b7fe-79b03f7f39aa.png)

**Comment by mickael-menu** [comment]
ShadowVim doesn't draw anything on the screen and relies on the native Xcode's selection to simulate the Vim block caret. Therefore it's not something we can do, as Xcode highlights the current line only when there are no active text selection.

Personally I changed the **Selection color** in my Xcode color scheme to make the caret more visible. I chose this green as I found it the most readable with all the colors in the default Dark theme: [#397C53](https://color-hex.org/color/397c53).

<img width="830" alt="Screenshot 2023-03-23 at 12 44 12" src="https://user-images.githubusercontent.com/58686775/227194504-d560552a-a511-45e9-8e5a-10be72acb6e6.png">

That being said, I think it might be possible to hijack the native Xcode Vim mode to use it alongside ShadowVim. This is really not a priority though and I have a lot of things to tackle first, so I wouldn't wait for it.
